editor/editor_properties.cpp

```diff
@@ -1864,6 +1864,13 @@ EditorPropertyColor::EditorPropertyColor() {
 	picker->set_flat(true);
 	picker->connect("color_changed", this, "_color_changed");
 	picker->connect("popup_closed", this, "_popup_closed");
+
+	// get default color picker mode from editor settings
+	int default_color_mode = EDITOR_GET("interface/inspector/default_color_picker_mode");
+	if (default_color_mode == 1)
+		picker->get_picker()->set_hsv_mode(true);
```

Comment:
I wonder if this has any performance impacts. ColorPickerButton doesn't create the picker until it is needed but calling `get_picker()` will initialize one all the time.
https://github.com/godotengine/godot/blob/4f6eb3610f70b2a73a19dd8b92fb03c4739ef25b/scene/gui/color_picker.cpp#L989-L994
The comment even mentions that it was done this way because of the property inspector.

Reply:
I could do it in 

> ColorPickerButton->get_picker()

 but that would mean going outside the editor which was the case in the earlier commit.

Reply:
Maybe ColorPickerButton should emit a signal once the ColorPicker is ready? Then you can connect to that.

Reply:
@merumelu That is indeed a better solution.
Made changes accordingly. Please review.